src/main.c

```diff
@@ -875,17 +744,77 @@ static int send_event(enum zl_event_t event_type, void *event_data) {
   return 0;
 }
 
+static int get_component_list(char *buf, size_t buf_size) {
+  char command[4*PATH_MAX];
+  snprintf (command, sizeof(command), "%s/bin/internal/get-launch-components.sh -c %s -r %s",
+    zl_context.root_dir, zl_context.instance_dir, zl_context.root_dir);
+  INFO("about to run command %s\n", command);
+  FILE *fp = popen(command, "r");
+  if (!fp) {
+    ERROR("unable to get start components - %s\n", strerror(errno));
+    return -1;
+  }
+  char *line = fgets(buf, buf_size - 1, fp);
+  fclose(fp);
+  if (!line) {
+    ERROR("failed to read start component list - %s\n", strerror(errno));
+    return -1;
+  }
+  int len = strlen(line);
+  for (int i = len - 1; i >= 0; i--) {
+    if (line[i] != ' ' && line[i] != '\n' && line[i] != ',') {
+      break;
+    }
+    line[i] = '\0';
+  }
+  if (strlen(line) == 0) {
+    ERROR("start component list is empty\n");
+    return -1;
+  }
+  INFO("start component list: '%s'\n", line);
+  return 0;
+}
+
+static int prepare_workspace() {
+  char command[PATH_MAX];
+  INFO("about to prepare zowe workspace\n");
+  const char *script = "bin/internal/prepare-workspace-for-launcher.sh";
+  snprintf (command, sizeof(command), "%s/%s", zl_context.root_dir, script);
+  int rc;
+  if ((rc = system(command)) != 0) {
+    ERROR("failed to prepare zowe workspace: %s ended with code %d\n", command, rc);
+    return -1;
+  }
+  INFO("zowe workspace prepared successfully\n");
+  return 0;
+}
+
 int main(int argc, char **argv) {
 
   INFO("Zowe Launcher starting\n");
 
   zl_config_t config = read_config(argc, argv);
 
-  if (init_context(&config)) {
+  if (init_context(argc, argv, &config)) {
     exit(EXIT_FAILURE);
   }
+  
+  char buf[512];
```

Comment:
Can you use a more descriptive name? `comp_buffer`?

Reply:
Sure.